include/mbedtls/config_psa.h

```diff
@@ -54,25 +54,143 @@ extern "C" {
 #define MBEDTLS_ECDSA_C
 #define MBEDTLS_HMAC_DRBG_C
 #define MBEDTLS_MD_C
-#endif /* MBEDTLS_PSA_ACCEL_ALG_DETERMINISTIC_ECDSA */
+#endif /* !MBEDTLS_PSA_ACCEL_ALG_DETERMINISTIC_ECDSA */
 #endif /* PSA_WANT_ALG_DETERMINISTIC_ECDSA */
 
+#if defined(PSA_WANT_ALG_ECDH)
+#if !defined(MBEDTLS_PSA_ACCEL_ALG_ECDH)
+#define MBEDTLS_PSA_BUILTIN_ALG_ECDH                    1
+#define MBEDTLS_ECDH_C
+#define MBEDTLS_ECP_C
+#define MBEDTLS_BIGNUM_C
+#endif /* !MBEDTLS_PSA_ACCEL_ALG_ECDH */
+#endif /* PSA_WANT_ALG_ECDH */
+
+#if defined(PSA_WANT_ECC_PUBLIC_KEY) || defined(PSA_WANT_ECC_KEY_PAIR)
```

Comment:
If I want ECC_PUBLIC_KEY but not ECC_KEY_PAIR and I have an accelerator for ecc public keys but not for key pairs I have:
PSA_WANT_ECC_PUBLIC_KEY defined
PSA_WANT_ECC_KEY_PAIR NOT defined
MBEDTLS_PSA_ACCEL_ECC_PUBLIC_KEY defined
MBEDTLS_PSA_ACCEL_ECC_KEY_PAIR NOT defined
Thus as MBEDTLS_PSA_ACCEL_ECC_KEY_PAIR is NOT defined, MBEDTLS_PSA_BUILTIN_ECC_KEY_PAIR is defined which not really what we want I think as PSA_WANT_ECC_KEY_PAIR is not defined.
The logic seems thus wrong here and  PSA_WANT_ECC_PUBLIC_KEY and PSA_WANT_ECC_KEY_PAIR should be treated separately: 
```c
#if defined(PSA_WANT_ECC_PUBLIC_KEY)
#if !defined(MBEDTLS_PSA_ACCEL_ECC_PUBLIC_KEY)
#define MBEDTLS_PSA_BUILTIN_ECC_PUBLIC_KEY              1
#define MBEDTLS_ECP_C
#define MBEDTLS_BIGNUM_C
#endif /* !MBEDTLS_PSA_ACCEL_ECC_PUBLIC_KEY */
#endif /* PSA_WANT_ECC_PUBLIC_KEY */

#if defined(PSA_WANT_ECC_KEY_PAIR)
#if !defined(MBEDTLS_PSA_ACCEL_ECC_KEY_PAIR)
#define MBEDTLS_PSA_BUILTIN_ECC_KEY_PAIR              1
#define MBEDTLS_ECP_C
#define MBEDTLS_BIGNUM_C
#endif /* !MBEDTLS_PSA_ACCEL_ECC_KEY_PAIR */
#endif /* PSA_WANT_ECC_KEY_PAIR */
```

There is the same kind of pattern below: 
#if defined(PSA_WANT_ALG_RSA_OAEP) || defined(PSA_WANT_ALG_RSA_PSS)
#if defined(PSA_WANT_ALG_RSA_PKCS1V15_CRYPT) || defined(PSA_WANT_ALG_RSA_PKCS1V15_SIGN)
#if defined(PSA_WANT_KEY_TYPE_RSA_KEY_PAIR) || defined(PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY)